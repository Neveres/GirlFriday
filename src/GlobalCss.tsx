import React from 'react'
import { css, Global } from '@emotion/react'

export const GlobalCss = () => (
  <Global
    styles={css`
      * {
        font-family: ${fontFamily.ubuntu};
        font-style: normal;
        color: ${colors.textWhite};
      }

      body {
        margin: 0;
        background-color: ${colors.bgPrimary};
      }

      #root {
        min-height: 100vh;
        position: relative;
      }

      .mobile-page-title {
        font-weight: ${fontWeight.lighter};
        font-size: ${fontSize.xxLarge};
        line-height: 150%;
        margin-bottom: 24px;
      }
    `}
  />
)

export const fontSize = {
  xxxLarge: '48px',
  xxLarge: '24px',
  xLarge: '16px',
  large: '14px',
  medium: '13px',
  small: '12px',
  xSmall: '11px',
}

export const fontWeight = {
  bolder: 700,
  bold: 600,
  lighter: 400,
}

export const fontFamily = {
  ubuntu: 'Ubuntu',
}

export const colors = {
  bgPrimary: '#181818',
  bgSecondary: '#1B1B1B',
  textWhite: '#ffffff',
  textGray: '#b2b2b2',
}
