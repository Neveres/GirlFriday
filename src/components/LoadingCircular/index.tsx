export { default, default as LoadingCircular } from './LoadingCircular'
