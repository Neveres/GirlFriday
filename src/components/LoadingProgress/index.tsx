export { default, default as LoadingProgress } from './LoadingProgress'
