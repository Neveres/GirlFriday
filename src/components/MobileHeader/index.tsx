export { default, default as MobileHeader } from './MobileHeader'
