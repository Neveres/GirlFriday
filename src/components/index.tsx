export { NavBar } from './NavBar'
export { AppBody } from './AppBody'
export { AppContext } from './AppContext'
export { Input } from './Input'
export { Button } from './Button'
export { Slider } from './Slider'
