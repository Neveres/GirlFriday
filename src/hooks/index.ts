export * from './useInnerWidth'
export * from './useTags'
export * from './useUsers'
export * from './useStorage'
export * from './useFriends'
