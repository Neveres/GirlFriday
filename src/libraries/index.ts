export * from './reducer'
export * from './action'
export * from './httpClient'
export * from './storage'
