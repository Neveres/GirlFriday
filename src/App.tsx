import React, { Suspense, useReducer, useMemo } from 'react'
import { useLocation } from 'react-router'
import { NavBar, AppBody, AppContext } from 'src/components'
import { Routes } from 'src/Routes'
import { useDevice } from 'src/hooks'
import { reducer, SET_MOBILE, SET_SEARCH_PARAMETERS } from 'src/libraries'
import { GlobalCss } from './GlobalCss'

const initialState = {
  isMobile: false,
  searchParameters: {
    page: 1,
    pageSize: 10,
    keyword: '',
  } as GirlFriday.SearchParameters,
}

const App = () => {
  const { pathname } = useLocation()

  const [state, dispatch] = useReducer(reducer, initialState)
  const actions = useMemo(
    () => ({
      setSearchParameters(searchParameters: GirlFriday.SearchParameters) {
        dispatch({ type: SET_SEARCH_PARAMETERS, ...state, searchParameters })
      },
      setMobile(isMobile: boolean) {
        dispatch({ type: SET_MOBILE, ...state, isMobile })
      },
    }),
    [state],
  )
  useDevice({ setMobile: actions.setMobile })
  const valueOfAppContext = useMemo(
    () => ({
      state,
      actions,
    }),
    [state, actions],
  )

  return (
    <>
      <AppContext.Provider value={valueOfAppContext}>
        <NavBar pathname={pathname} />

        <AppBody pathname={pathname}>
          <Suspense fallback={<div>Loading...</div>}>
            <Routes />
          </Suspense>
        </AppBody>
      </AppContext.Provider>

      <GlobalCss />
    </>
  )
}

export default App
