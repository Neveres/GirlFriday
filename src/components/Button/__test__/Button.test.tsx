import React from 'react'
import { create } from 'react-test-renderer'
import { Button } from 'src/components'

describe('Button', () => {
  const props = {
    onClick: jest.fn(),
    text: 'text',
    layout: 'desktop',
  }

  test('should render well under desktop layout', () => {
    const testRenderer = create(<Button {...props} />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })

  test('should render well under mobile layout', () => {
    props.layout = 'mobile'
    const testRenderer = create(<Button {...props} />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })

  test('type "contained" should render well', () => {
    const testRenderer = create(<Button {...props} type="contained" />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })

  test('type "outlined" should render well', () => {
    const testRenderer = create(<Button {...props} type="outlined" />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
