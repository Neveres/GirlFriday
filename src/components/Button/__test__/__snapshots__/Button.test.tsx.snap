// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render well under desktop layout 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium mobile css-1qtm4ck-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  text
</button>
`;

exports[`Button should render well under mobile layout 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium mobile css-qrm0fn-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  text
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`Button type "contained" should render well 1`] = `
<div
  className="css-385uyt-Contained"
>
  <span
    className="button-text"
  >
    text
  </span>
</div>
`;

exports[`Button type "outlined" should render well 1`] = `
<div
  className="css-2qsxy8-Outlined"
>
  <span
    className="button-text"
  >
    text
  </span>
</div>
`;
